### PR TITLE
Cadence Cookbook Updates - Functional MVP

### LINKED ISSUE #11: [FEATURE]  New UI release
### Issue to be solved

- Updated deployment/build scripts for new codebase. Create Github actions for deployment, integrate with MVP Github action(s) to run tests before the build. 
- Merge new front-end code into master branch. 
- Transfer ownership of recipe sub-repos to Flow.
- Deploy to existing environment (unknown?). 
- New UI released live.

### Suggest A Solution

_No response_

### What are you currently working on that this is blocking?

_No response_

### LINKED ISSUE #12: [FEATURE]  Recipe content derived from Github subrepos
### Issue to be solved

-Creating a repo for each recipe
- Moving code out to Cadence files/text out to txt files, and creating a reference to recipe submodule in the parent repo.
- Assigning parent modules, labels/categories to onboarded. 
- Assigning a chronological ordering to recipes. 
- Total 22 recipes (excluding 2 KittyItems recipes which will be removed from new Cookbook).

### Suggest A Solution

_No response_

### What are you currently working on that this is blocking?

_No response_

### LINKED ISSUE #13: [FEATURE] Testing configuration
### Issue to be solved

- E2E and unit testing setup on front-end code
- MVP of Cadence testing running on individual recipe code in subrepos
- Tests run as part of a Github action in parent repo

### Suggest A Solution

_No response_

### What are you currently working on that this is blocking?

_No response_

### LINKED ISSUE #14: [FEATURE] Static indexing of recipe pages, SEO improvements
### Issue to be solved

- Each recipe indexed as its own page with unique URL; move recipe content out of popup modal
- Title, meta, keywords, OpenGraph and Google Schema, sitemap, setup for all recipe pages and homepage

### Suggest A Solution

_No response_

### What are you currently working on that this is blocking?

_No response_

### LINKED ISSUE #15: [FEATURE] Improved contribution flow
### Issue to be solved

- Updated documentation for new process of contributing to the Cookbook (how contribution repo and PR should be formatted, what fields are supported/required). 

### Suggest A Solution

_No response_

### What are you currently working on that this is blocking?

_No response_

### PR DESCRIPTION
**Pull request by lealobanov**
This pull request implements functionality for the following issues:
- Closes #11 
- Closes #12 
- Closes #13
- Closes #15 
- Closes #14, as well as #8

### CODE REVIEW

**Review by Aliserag — APPROVED** [review]

**Review by franklywatson — APPROVED** [review]

**Comment by bthaile** [comment]
Is there a vercel preview we can look at?

I found https://cadence-cookbook-axno3rxr2-onflow.vercel.app/

it's the latest deployment on branch `[lealobanov:new](https://github.com/lealobanov/cadence-cookbook/tree/new)`

It seems a bit off. Might be a cache issues. images did end up loading.
image

A few small issues:

The whole card should be clickable not just the title:


UI elements being center justified looks off and not clean, imo 
just having the text line up with left of code would work. 
image

Need to remove `margin:0` for `p` tag. This will allow paragraph breaks. The brick of text doesn't look good.
image

File: src/data/recipeUtils.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      return e.slug;
+    })
+    .indexOf(slug);
+
+  const totalRecipes = recipes.length;
+
+  const proportionComplete = (thisRecipeIndex + 1) / totalRecipes;
+
+  return proportionComplete;
+}
```
Comment (nialexsan) [inline src/data/recipeUtils.js]: do these functions need to be async?
it seems that all functions are synchronous

Comment (lealobanov) [reply src/data/recipeUtils.js]: Hi @nialexsan , I initially marked them all as async assuming we might load in some external data (e.g. videos, embedded runtime) in the future. Will remove for now and we can add async keyword back later if needed. Thanks!

**Comment by lealobanov** [comment]
@bthaile , noted on Cards not being clickable, working on adding that now.

On my side I see the code blocks aligned to the width of the text; I think it might have to do with a different screen size, will play around with it to see if I can have consistent padding and otherwise will left-justify.

Thanks!

Screenshot 2023-12-02 at 01 43 56

**Comment by lealobanov** [comment]
Re. spacing in paragraph, the text is currently loaded as markdown as it is coming from txt files. I think the line breaks might need to be added in the txt file itself - will double check.

**Comment by lealobanov** [comment]
@bthaile @nialexsan  Hi all, I have made some adjustments yesterday and pushed to the latest Vercel deployment: https://cadence-cookbook-6vshgor66-onflow.vercel.app. Please have a look when you get a chance. Thank you!

**Review by nialexsan — APPROVED** [review]

**Pull request merged**